packages/grafbase-sdk/src/auth.ts

```diff
@@ -0,0 +1,157 @@
+import { JWKSAuth } from './auth/jwks'
+import { JWTAuth } from './auth/jwt'
+import { OpenIDAuth } from './auth/openid'
+
+export type AuthProvider = OpenIDAuth | JWTAuth | JWKSAuth
+export type AuthRuleF = (rules: AuthRules) => any
+export type AuthOperation =
+  | 'get'
+  | 'list'
+  | 'read'
+  | 'create'
+  | 'update'
+  | 'delete'
+
+export class AuthGroups {
+  groups: string[]
+
+  constructor(groups: string[]) {
+    this.groups = groups
+  }
+
+  public toString(): string {
+    const groups = this.groups.map((g) => `"${g}"`).join(', ')
+    return `groups, groups: [${groups}]`
+  }
+}
+
+export type AuthStrategy = 'private' | 'owner' | AuthGroups
+
+export class AuthRule {
+  strategy: AuthStrategy
+  operations: AuthOperation[]
+
+  constructor(strategy: AuthStrategy) {
+    this.strategy = strategy
+    this.operations = []
+  }
+
+  public get(): AuthRule {
+    return this.operation('get')
+  }
+
+  public list(): AuthRule {
+    return this.operation('list')
+  }
+
+  public read(): AuthRule {
+    return this.operation('read')
+  }
+
+  public create(): AuthRule {
+    return this.operation('create')
+  }
+
+  public update(): AuthRule {
+    return this.operation('update')
+  }
+
+  public delete(): AuthRule {
+    return this.operation('delete')
+  }
+
+  public all() {
```

Comment:
Do we need this? If no operation rule is specified then it'll default to all in the backend afaik.

Reply:
That's not to say I don't prefer being more explicit.

Reply:
No we don't.